From 1678ccc82e17040e84d4a38d163a5d341ebd35d0 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Fri, 16 Aug 2024 05:51:43 +0800
Subject: [PATCH 18/19] Build LoongArch64 LSX optimizations in libpng
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

These files were added in Chromium Change Number 5454197, but not referenced in BUILD.gn.

Update BUILD.gn to compile them when appropriate.

Bug: 356038456
Change-Id: I2d15071261a3785184c5fa7c26ea26b182d4ba6c
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5740787
Reviewed-by: Rick Byers <rbyers@chromium.org>
Commit-Queue: Rick Byers <rbyers@chromium.org>
Auto-Submit: Kexy Biscuit (百合ヶ咲るる / 被子饼) <kexybiscuit@gmail.com>
Reviewed-by: Leon Scroggins <scroggo@google.com>
Cr-Commit-Position: refs/heads/main@{#1338820}
---
 AUTHORS                     | 2 ++
 third_party/libpng/BUILD.gn | 6 ++++++
 2 files changed, 8 insertions(+)

diff --git a/AUTHORS b/AUTHORS
index ad27653d70..529418bf08 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -786,6 +786,8 @@ Ketan Goyal <ketan.goyal@samsung.com>
 Kevin Gibbons <bakkot@gmail.com>
 Kevin Lee Helpingstine <sig11@reprehensible.net>
 Kevin M. McCormick <mckev@amazon.com>
+Kexy Biscuit <kexybiscuit@aosc.io>
+Kexy Biscuit <kexybiscuit@gmail.com>
 Keyou <qqkillyou@gmail.com>
 Khasim Syed Mohammed <khasim.mohammed@linaro.org>
 Khem Raj <raj.khem@gmail.com>
diff --git a/third_party/libpng/BUILD.gn b/third_party/libpng/BUILD.gn
index 81648e3b2a..3116ea0e77 100644
--- a/third_party/libpng/BUILD.gn
+++ b/third_party/libpng/BUILD.gn
@@ -4,6 +4,7 @@
 
 import("//build/config/arm.gni")
 import("//build/config/mips.gni")
+import("//build/config/loongarch64.gni")
 
 if (build_with_chromium) {
   import("//testing/libfuzzer/fuzzer_test.gni")
@@ -91,6 +92,11 @@ source_set("libpng_sources") {
       "mips/filter_msa_intrinsics.c",
       "mips/mips_init.c",
     ]
+  } else if (current_cpu == "loong64" && loongarch64_use_lsx) {
+    sources += [
+      "loongarch/filter_lsx_intrinsics.c",
+      "loongarch/loongarch_lsx_init.c",
+    ]
   }
 
   configs -= [ "//build/config/compiler:chromium_code" ]
-- 
2.46.0

